lib/puppet/util/windows/adsi.rb

```diff
@@ -162,6 +162,13 @@ def commit
       begin
         native_user.SetInfo unless native_user.nil?
       rescue Exception => e
+        if e.message =~ /8007089A/m
```

Comment:
So in debugging, I see that this error is actually a `WIN32OLERuntimeError` type, so I would rather us be a bit more explicit about catching that type.